### ISSUE TITLE
CMake fails if `find_package(ucx)` happens more than once

### ISSUE THREAD

**Issue by pentschev** [openucx/ucx#9614]
### Describe the bug
CMake fails if a project attempts to `find_package(ucx)` more than once in the same CMake project:

```
CMake Error at /datasets/pentschev/miniconda3/envs/rn-240118/lib/cmake/ucx/ucx-targets.cmake:10 (add_library):
  add_library cannot create imported target "ucx::ucs" because another target
  with the same name already exists.
Call Stack (most recent call first):
  /datasets/pentschev/miniconda3/envs/rn-240118/lib/cmake/ucx/ucx-config.cmake:7 (include)
  CMakeLists.txt:4 (find_package)


CMake Error at /datasets/pentschev/miniconda3/envs/rn-240118/lib/cmake/ucx/ucx-targets.cmake:17 (add_library):
  add_library cannot create imported target "ucx::ucp" because another target
  with the same name already exists.
Call Stack (most recent call first):
  /datasets/pentschev/miniconda3/envs/rn-240118/lib/cmake/ucx/ucx-config.cmake:7 (include)
  CMakeLists.txt:4 (find_package)


CMake Error at /datasets/pentschev/miniconda3/envs/rn-240118/lib/cmake/ucx/ucx-targets.cmake:24 (add_library):
  add_library cannot create imported target "ucx::uct" because another target
  with the same name already exists.
Call Stack (most recent call first):
  /datasets/pentschev/miniconda3/envs/rn-240118/lib/cmake/ucx/ucx-config.cmake:7 (include)
  CMakeLists.txt:4 (find_package)
```

### Steps to Reproduce
- Create a new `CMakeLists.txt` with the content below and run `cmake .`.

```cmake
project(ucx_proj)

find_package(ucx)
find_package(ucx)
```

- UCX 1.15.0 and latest master @ https://github.com/openucx/ucx/commit/d35d3057dbab335c8dc58d3d7e32851dbe153c3c

**Comment by pentschev** [comment]
@vyasr could you comment here? I think you know how to fix it and have a proposal for that.

**Comment by vyasr** [comment]
Yup I uncovered this while developing a package that brings in ucx as a transitive dependency. The issue is that the [ucx-targets.cmake.in template](https://github.com/openucx/ucx/blob/master/cmake/ucx-targets.cmake.in) does not check for the existence of targets before running `add_library`, which makes the multiple inclusion unsafe. What we want is to check that all the expected targets have already been created, and if so return early. To be safe we can also error if a partial subset of the targets exist.

I'm happy to open a PR with this patch if the developers agree with it.

**Comment by vyasr** [comment]
I've submitted a proposed patch in #9622

**Comment by tvegas1** [comment]
fixing in ucx with #9696

**Comment by pentschev** [comment]
> is #9696 it needed in previous versions?

Thanks @tvegas1 ! Yes, could it be backported to 1.16? That would simplify a lot for us in the near-term.

**Comment by tvegas1** [comment]
v1.16.x backport: #9702

**Comment by pentschev** [comment]
Thanks @tvegas1 !

**Comment by tvegas1** [comment]
merged in master and v1.16.x

**Comment by vyasr** [comment]
Thanks for getting this fixed!
